staging/src/k8s.io/api/bulk/v1alpha1/types.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+
+// BulkRequest contains a single request to bulk api.
+type BulkRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Opaque identifier used to match requests and responses.
+	RequestID string `json:"requestId" protobuf:"bytes,1,name=requestId"`
```

Comment:
I think the semantics of this field would be significantly better if we made this object not a union and instead had e.g. StartWatchRequest, StartWatchListRequest, StopWatchRequest, ListRequest, etc., messages.

Because I would like to document "RequestID must be unique" but that's not true if you're sending a StopWatch request that references a past Watch request.

Reply:
> Because I would like to document "RequestID must be unique" but that's not true if you're sending a StopWatch request that references a past Watch request.

ReqeustID still should be unique in this case. Referring is performed by 'WatchID' instead of RequestID.

> I think the semantics of this field would be significantly better if we made this object not a union and instead had e.g. StartWatchRequest, StartWatchListRequest, StopWatchRequest, ListRequest, etc., messages.

I'm don't fully understand what will change with 'RequestID' after switching to separate top-level request/response messages. IMO semantic of RequestID will not change at all.